runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/operation/SdkOperationExecution.kt

```diff
@@ -29,39 +34,104 @@ import aws.smithy.kotlin.runtime.io.middleware.decorate as decorateHandler
-    // are described with their intended use, nothing prevents e.g. registering
-    // something in "initialize" that acts on the output type though.
-
     /**
-     * Prepare the input [Request] (or finalize the [Response]) e.g. set any default parameters if needed
+     * **Request Middlewares**: Prepare the input [Request] (e.g. set any default parameters if needed) before
+     * serialization
+     *
+     * **Response Middlewares**: Finalize the [Response] before returning it to the caller
      */
     public val initialize: Phase<OperationRequest<Request>, Response> = Phase<OperationRequest<Request>, Response>()
 
     /**
-     * Modify the outgoing HTTP request
+     * **Request Middlewares**: Modify the outgoing HTTP request. This phase runs BEFORE the retry loop and
+     * is suitable for any middleware that only needs to run once and does not need to modify the outgoing request
+     * per/attempt.
      *
      * At this phase the [Request] (operation input) has been serialized to an HTTP request.
+     *
+     * **Response Middlewares**: Modify the output after deserialization
      */
     public val mutate: Phase<SdkHttpRequest, Response> = Phase<SdkHttpRequest, Response>()
 
     /**
-     * Last chance to intercept before requests are sent (e.g. signing, retries, etc).
-     * First chance to intercept after deserialization.
+     * **Request Middlewares**: Modify the outgoing HTTP request. This phase is conceptually the same as [mutate]
+     * but it runs on every attempt. Each attempt will not see modifications made by previous attempts.
+     *
+     * **Response Middlewares**: Modify the output after deserialization on a per/attempt basis.
      */
-    public val finalize: Phase<SdkHttpRequest, Response> = Phase<SdkHttpRequest, Response>()
+    public val onEachAttempt: Phase<SdkHttpRequest, Response> = Phase<SdkHttpRequest, Response>()
 
     /**
-     * First chance to intercept before deserialization into operation output type [Response].
+     * **Request Middlewares**: First chance to intercept after signing (and last chance before the final request is sent).
+     *
+     * **Response Middlewares**: First chance to intercept after the raw HTTP response is received and before deserialization
+     * into operation output type [Response].
      */
     public val receive: Phase<SdkHttpRequest, HttpCall> = Phase<SdkHttpRequest, HttpCall>()
+
+    /**
+     * The [HttpSigner] to sign the request with
+     */
+    // FIXME - this is temporary until we refactor identity/auth APIs
+    public var signer: HttpSigner = HttpSigner.Anonymous
+
+    /**
+     * The retry strategy to use. Defaults to [StandardRetryStrategy]
+     */
+    public var retryStrategy: RetryStrategy = StandardRetryStrategy()
+
+    /**
+     * The retry policy to use. Defaults to [StandardRetryPolicy]
+     */
+    public var retryPolicy: RetryPolicy<Response> = StandardRetryPolicy.Default
```

Comment:
**Question**: Do these deserve `// FIXME` comments as well? Or are `public var`s the likely-permanent way middleware/interceptors will modify retry config?

Reply:
I had planned on these being more permanent. Middleware and interceptors can't really modify these. Plugins likely will be able to but not sure what that looks like. 

Open to discussion though.